src/catalog/src/builtin.rs

```diff
@@ -5659,12 +5659,33 @@ materialized_views AS (
     LEFT JOIN mz_internal.mz_compute_hydration_statuses h
         ON (h.object_id = i.id)
 ),
+sources_maintained_by_dataflows AS (
+    SELECT id, cluster_id
+    FROM mz_sources
+    WHERE type IN ('kafka', 'postgres', 'load-generator')
```

Comment:
This seems like something that we might forget to update, maybe we could add a column to the source or something?

Reply:
Maybe it'd be safer to negate the condition? `NOT IN ('webhook')`? Assuming that it'll be rarer that we add sources not backed by dataflows.

Reply:
I like the `NOT IN ('webhook')` idea. Webhook sources are strange because they have a `cluster_id` but are not backed by a dataflow. There are other source types (`log`, `source`, `progress`), but these all don't have a `cluster_id`, so checking for "`cluster_id` and not webhook" should work.